monitoring/uss_qualifier/reports/tested_requirements.py

```diff
@@ -189,22 +238,50 @@ def _sort_breakdown(breakdown: TestedBreakdown) -> None:
             requirement.scenarios.sort(key=lambda s: s.name)
 
 
+def _populate_breakdown_with_req_set(
+    breakdown: TestedBreakdown, req_set: Set[RequirementID]
+) -> None:
+    for req_id in req_set:
+        package_id = req_id.package()
+        matches = [p for p in breakdown.packages if p.id == package_id]
+        if matches:
+            tested_package = matches[0]
+        else:
+            tested_package = TestedPackage(
+                id=package_id, name=package_id, requirements=[]
+            )
+            breakdown.packages.append(tested_package)
+
+        short_req_id = req_id.split(".")[-1]
+        matches = [r for r in tested_package.requirements if r.id == short_req_id]
+        if matches:
+            tested_requirement = matches[0]
+        else:
+            tested_requirement = TestedRequirement(id=short_req_id, scenarios=[])
+            tested_package.requirements.append(tested_requirement)
```

Comment:
This looks suspicious to me, shouldn't this be the following?
```suggestion
        if matches:
            tested_package = matches[0]
        else:
            tested_package = TestedPackage(
                id=package_id, name=package_id, requirements=[]
            )
        breakdown.packages.append(tested_package)

        short_req_id = req_id.split(".")[-1]
        matches = [r for r in tested_package.requirements if r.id == short_req_id]
        if matches:
            tested_requirement = matches[0]
        else:
            tested_requirement = TestedRequirement(id=short_req_id, scenarios=[])
        tested_package.requirements.append(tested_requirement)
```

Reply:
No, we're looking for an existing package or adding a new one.  If there's already a package, we don't want to add that existing package to `breakdown.packages` again as doing so would duplicate the package.

Then, we're looking for an existing requirement or adding a new one.  If there's already a requirement, we don't want to add that requirement to `tested_package.requirements` again as doing so would duplicate the requirement.

Basically, we only want to add a package/requirement/scenario/case/step/check once and then mutate the existing instance each time after adding it that once.